File: Leetcode-Problems/Isomorphic Strings.cpp
class Solution
{
public:
    bool isIsomorphic(string s, string t)
    {
        unordered_map<char, char> mpS;
        unordered_map<char, char> mpT;

        for (int i = 0; i < s.length(); i++)
        {
            if (mpS.count(s[i]) && mpS[s[i]] != t[i])
                return false;
            if (mpT.count(t[i]) && mpT[t[i]] != s[i])
                return false;

            mpS[s[i]] = t[i];
            mpT[t[i]] = s[i];
        }
        return true;
    }
};

File: Leetcode-Problems/Isomorphic Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> mpS = new HashMap<>();
        HashMap<Character, Character> mpT = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            if ((mpS.containsKey(s.charAt(i)) && mpS.get(s.charAt(i)) != t.charAt(i))
                    || (mpT.containsKey(t.charAt(i)) && mpT.get(t.charAt(i)) != s.charAt(i))) {
                return false;
            }
            mpS.put(s.charAt(i), t.charAt(i));
            mpT.put(t.charAt(i), s.charAt(i));
        }
        return true;
    }
}